daemon/tests/test.rs

```diff
@@ -0,0 +1,396 @@
+/*
+Copyright (c) 2022 VMware, Inc.
+SPDX-License-Identifier: MIT
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+ */
+
+// Derived from tests/daemon.at in Open vSwitch, with the following license:
+/*
+Copyright (c) 2009, 2010, 2011, 2012, 2013, 2014, 2015 Nicira, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at:
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use anyhow::{anyhow, Result, Context};
+use std::process::{Child, Command, ExitStatus};
+use std::path::{Path, PathBuf};
+use std::os::raw::c_int;
+use std::io::ErrorKind;
+use std::sync::Mutex;
+
+fn parent_pid(pid: libc::pid_t) -> Result<libc::pid_t> {
+    let output = String::from_utf8(Command::new("ps")
```

Comment:
is this also posix?

Reply:
shouldn't this also be `#cfg("linux")`?

Reply:
This *is* POSIX, see https://pubs.opengroup.org/onlinepubs/9699919799/utilities/ps.html. I know that's hard to believe. I'll add a comment.